cmd/wallet-sdk-gomobile/did/wellknown.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package did
+
+import (
+	"errors"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/didconfig"
+
+	"github.com/trustbloc/wallet-sdk/cmd/utilities/gomobilewrappers"
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
+	"github.com/trustbloc/wallet-sdk/pkg/common"
+	goapiwellknown "github.com/trustbloc/wallet-sdk/pkg/did/wellknown"
+)
+
+// A ValidationResult is the type returned from the ValidateService method.
+// IsValid indicates if the given DID passed the service validation check, and ServiceURL indicates the URL of
+// that service.
+type ValidationResult struct {
+	IsValid    bool
+	ServiceURL string
+}
+
+// ValidateService validates the given DID's service against its well-known DID configuration.
+// It returns a ValidationResult.
+// The DID document must specify only a single service. If there are multiple URLs for a given service, only the
+// first will be checked.
+func ValidateService(did string, resolver api.DIDResolver) (*ValidationResult, error) {
```

Comment:
Should this be names ValidateWellKnownDIDConfig as that's the actual validation? There can be multiple services, but this specific validation is for Well Known DID Config.

Reply:
I think `ValidateWellKnownDIDConfig` would be somewhat misleading, as the main point of this function isn't validating the well-known DID config itself, but rather is validating the service in the DID doc _against_ its well-known DID config. (The function in Aries that I call does happen to also validate the well-known DID config along the way, but I think that's just a piece of the overall picture)

Reply:
There could be multiple services inside a DID Document. We are validating services with `"type": "LinkedDomains"` - https://identity.foundation/.well-known/resources/did-configuration/#example-7.

Reply:
agree with @rolsonquadras 
`VaidateService` is too generic name, should be called `ValidateLinkedDomain, ValidateWellKnown or ValidateWellKnownConfig`

Reply:
`ValidateLinkedDomains` sounds like a better name to me. I've renamed it to that (and also added in a check to ensure the service is of type `LinkedDomains`)